### PR TITLE
EVG-13502: undo revert, write DependenciesMetTime to db if it is being calculated for the first time

### PR DESCRIPTION
**Pull request by hhoke**
This was reverted due to failing smoke tests on the waterfall, but that [appears to have been resolved](https://spruce.mongodb.com/version/60181f5c9ccd4e13c00f2e78/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

revert: https://github.com/evergreen-ci/evergreen/pull/4330
original PR: https://github.com/evergreen-ci/evergreen/pull/4321

### CODE REVIEW

**Comment by ablack12** [comment]
Did the smoke tests fail after your changes had been deployed, or after they had been committed? If the former, you'll want to deploy your changes to staging, and then run the smoke tests from there.

File: self-tests.yml [hunk 1]
```diff
@@ -544,6 +544,25 @@ tasks:
           target: revendor
       - func: run-make
         vars: { target: "test-repotracker" }
+  - name: dep-test
```
Comment (ablack12) [inline self-tests.yml:547]: Is this something we want to commit?

Comment (hhoke) [reply self-tests.yml:547]: I personally find it useful but if you think it's unneeded I'm happy to get rid of it.

**Comment by hhoke** [comment]
Smoke tests failed after changes were committed. We never deployed out of caution, though even before the revert smoke tests had run fine on subsequent commits.

File: self-tests.yml (hunk 1 above)
Comment (ablack12) [reply self-tests.yml:547]: Is there a way to re-write this as a unit test? That's more how we should be testing functionality, as opposed to running an empty test.

Comment (ablack12) [reply self-tests.yml:547]: Otherwise I would just remove it now that your testing is done.

File: model/task/task.go [hunk 2]
```diff
@@ -464,11 +463,17 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
...
-	t.DependenciesMetTime = latestTime
+	// this is not exact, but depTask.FinishTime is not always set in time to use that
+	t.DependenciesMetTime = time.Now()
+	err = UpdateOne(
+		bson.M{IdKey: t.Id},
+		bson.M{
+			"$set": bson.M{DependenciesMetTimeKey: t.DependenciesMetTime},
+		})
+	if err != nil {
+		return true, errors.Wrapf(err, "task.DependenciesMet() failed to update task '%s'", t.Id)
```
Comment (ablack12) [inline model/task/task.go:475]: Even though we return true with the error here, in the cases we use this function, we ignore the boolean value if there's an error. I think it might be better to just log the error, and still return true regardless, so that things can continue.

File: self-tests.yml (hunk 1 above)
Comment (hhoke) [reply self-tests.yml:547]: I wanted more for an end-to-end staging test for functionality that involves dependencies, so I'll just get rid of it and stick it in my notebook in case this is needed again.

**Review by ablack12 — APPROVED** [review]
Sorry, I had this comment queued up earlier but apparently didn't press submit. LGTM after that's fixed though.
